detox/src/realms/DetoxContext.js

```diff
@@ -0,0 +1,180 @@
+const funpermaproxy = require('funpermaproxy');
+
+const temporary = require('../artifacts/utils/temporaryPath');
+const { DetoxRuntimeError } = require('../errors');
+const { DetoxLogger, installLegacyTracerInterface } = require('../logger');
+const symbols = require('../symbols');
+
+const DetoxConstants = require('./DetoxConstants');
+
+const $cleanup = Symbol('cleanup');
+const $restoreSessionState = Symbol('restoreSessionState');
+const $sessionState = Symbol('restoreSessionState');
+const $status = Symbol('status');
+const $worker = Symbol('worker');
+
+class DetoxContext {
+  constructor() {
+    /** @type {DetoxInternals.DetoxStatus} */
+    this[$status] = 'inactive';
+
+    const _init = this[symbols.init].bind(this);
+    this[symbols.init] = async (opts) => {
+      this[$status] = 'init';
+      await _init(opts);
+      this[$status] = 'active';
+    };
+
+    const _cleanup = this[symbols.cleanup].bind(this);
+    this[symbols.cleanup] = async () => {
+      this[$status] = 'cleanup';
+      try {
+        await _cleanup();
+      } finally {
+        this[$status] = 'inactive';
+      }
+    };
+
+    this[symbols.getStatus] = this[symbols.getStatus].bind(this);
+    this[symbols.reportTestResults] = this[symbols.reportTestResults].bind(this);
+    this[symbols.resolveConfig] = this[symbols.resolveConfig].bind(this);
+    this[symbols.installWorker] = this[symbols.installWorker].bind(this);
+    this[symbols.uninstallWorker] = this[symbols.uninstallWorker].bind(this);
+
+    this[$sessionState] = this[$restoreSessionState]();
+
+    /**
+     * @type {DetoxLogger & Detox.Logger}
+     */
+    this[symbols.logger] = new DetoxLogger({
+      file: temporary.for.jsonl(`${this[$sessionState].id}.${process.pid}`),
+      userConfig: this[$sessionState].detoxConfig
+        ? this[$sessionState].detoxConfig.logger
+        : null,
+    });
+
+    this.log = this[symbols.logger].child({ cat: 'user' });
+    installLegacyTracerInterface(this.log, this);
+
+    /** @type {import('../DetoxWorker') | null} */
```

Comment:
Nope, maybe in some specific cases only. The syntax I use is currently the most universal one, it seems. 🤷‍♂️ 

<https://stackoverflow.com/a/52847569>

Reply:
If it were a public type from our TypeScript definitions than maybe it could work, but here I want it to acquire the shape of exactly `DetoxWorker.js` module.